server/channels/store/storetest/channel_store_categories.go

```diff
@@ -2413,3 +2423,251 @@ func testSidebarCategoryDeadlock(t *testing.T, rctx request.CTX, ss store.Store)
 
 	wg.Wait()
 }
+
+func testSidebarCategoryConcurrentAccess(t *testing.T, rctx request.CTX, ss store.Store, s SqlStore) {
+	if s.DriverName() == model.DatabaseDriverMysql {
+		t.Skip("This is known to fail on MySQL")
+	}
+
+	userID := model.NewId()
+	team := setupTeam(t, rctx, ss, userID)
+
+	numGoroutines := 40
```

Comment:
40 is too high to be run in CI. We can run into resource issues. I'd say 20 is still good to repro this.